MIGRATION.md

```diff
@@ -1,11 +1,8 @@
 # Migration Instructions for Major Releases
 
-## From `10.0.0` to `11.0.0`
-
- - The new minimum required Ruby version is `3.2.2`. Please make sure to upgrade your projects before upgrading to this version to avoid compatibility issues.
-
 ## From `9.0.0` to `10.0.0`
 
+ - The new minimum required Ruby version is `3.2.2`. Please make sure to upgrade your projects before upgrading to this version to avoid compatibility issues.
```

Comment:
I think the reason this was initially added under the `from `10.0.0` to `11.0.0` section is that we maybe considered that the Ruby version bump would be a breaking change?

I mean, technically it is, because by definition a breaking change is when the new version of a dependency will require you to make (breaking) changes to your project before being able to adopt it. And indeed, updating the `release-toolkit` in your client projects will require you to update your projects to Ruby 3 (or ensure they've migrated already, while before the Ruby version requirement was lower).

It's always an open debate if things like that (Ruby version, or thinks like the Swift Version or target_deployment_version bumps for iOS pods, etc) should be taken into account when considering breaking changes and the need for a major version bumps.

So I'm open to the other POV of this not being considered a breaking change, especially since we are the sole users of this lib (I don't think anyone outside a8c uses `release-toolkit` 😛 ) and we know that all our clients of that lib (all our apps projects) have migrated to Ruby 3 (except maybe Simplenote… yet, due to its maintenance mode?), we just need to all agree on our stance on this.

Reply:
> 💭 I think the reason this was initially added under the from 10.0.0to11.0.0` section is that we maybe considered that the Ruby version bump would be a breaking change?

I actually see it as a breaking change (as, for example, it will break a Gem import if you're on Ruby `2.x`), but in this case this move was a [mistake from my side](https://github.com/wordpress-mobile/release-toolkit/pull/517) as I just added the `11.0.0` entry but didn't realize we're still on `9.x`, going to 10.0.0 on the next release.

Reply:
Asked the team for their thoughts in https://a8c.slack.com/archives/C02KLTL3MKM/p1707414498911939 btw